src/Draco.FormatterEngine/Box.cs

```diff
@@ -0,0 +1,9 @@
+namespace Draco.Compiler.Internal.Syntax.Formatting;
+
+public class Box<T>(T value)
+{
+    protected T value = value;
+    public T Value => this.value;
+
+    public static implicit operator Box<T>(T value) => new(value);
+}
```

Comment:
Yeah I think nullability should be erased by box and we should provide a way to check, if it's been set. Practically a Task at that point.